### DISCUSSION TITLE
[UI] Template creation: a couple of questions

### DISCUSSION THREAD

**Discussion by alessandro-newzoo** [coder/coder#14258] [General] [answered]
Hello,

I have a couple of questions concerning the **UI**:

---

### 1. How does one export and import a template?

I bumped into this issue: https://github.com/coder/coder/issues/11687, in which [the last comment at the bottom](https://github.com/coder/coder/issues/11687#issuecomment-1968972862) says:
> Zip upload is supported by the UI now. I need to evaluate changing the template editor to use zip more carefully since this part of the code is a bit fragile at the moment.

and in this issue: https://github.com/coder/coder/issues/5313 the user is asking
> Template library & upload via the dashboard

and the last comment says (pointing to a piece of doc that I don't think is relevant to the subject):
> It is done and available under CODER_EXPERIMENTAL https://coder.com/docs/coder-oss/latest/admin/high-availability#setup

I'm confused as I can't find anything in the UI that allows me to upload anything.

---

### 2. Is it possible to create a template from scratch without having to pick a starter template first?

---

Can anybody please shed some light on these questions I have?

Thank you so much for this fantastic piece of software! 🖤

**Comment by matifali** [comment, marked as answer]
> 1. How does one export and import a template?

You can upload/import a new template using https://coder.example.com/templates/new. Currently Coder dos not have a way to download template tars from UI. CLI can be used `coder templates pull <TEMPLATE_NAME>`

> 2. Is it possible to create a template from scratch without having to pick a starter template first?

Yes. With both from
1. https://coder.example.com/templates/new
2. https://coder.example.com/templates/new?exampleId=scratch (This gives you a barebone template you can later edit to add new resources)

**Reply by alessandro-newzoo** [reply]
Thank you, I had no idea about those URLs!
May I ask why they're not accessible through the UI and are also not documented?

**Reply by matifali** [reply]
Oh coder.example.com is just a place holder. And you get these routes when you click the new template button under template page.

**Reply by alessandro-newzoo** [reply]
Oh yeah sorry I didn't mean that, I know it was a placeholder, what I meant is why isn't there a button in the UI to get to that URL?
I can't find the "New template" button you mentioned:

https://github.com/user-attachments/assets/7410d763-03d7-412c-b4f3-b0ad3e994855

I can only pick a Starter

**Reply by matifali** [reply]
That is interesting. It used to be there and isprobably a bug. Would you like to create a bug report that the new scratch template and upload template are not visible in UI?

**Reply by alessandro-newzoo** [reply]
ohhh that is so great to hear I thought I was going crazy 😄
Will do, thanks!

**Reply by matifali** [reply]
If you attach the video, that would be useful.

**Reply by alessandro-newzoo** [reply]
yep I've added it, it's here: https://github.com/coder/coder/issues/14323
thanks!

**Reply by matifali** [reply]
Thank you :)
